modules/virt-binding-devices-vfio-driver.adoc

```diff
@@ -0,0 +1,137 @@
+// Module included in the following assemblies:
+//
+// * virt/virtual_machines/advanced_vm_management/virt-configuring-pci-passthrough.adoc
+
+[id="virt-binding-devices-vfio-driver_{context}"]
+= Binding PCI devices to the VFIO driver
+To bind PCI devices to the VFIO driver, obtain the values for `vendor-ID` and `device-ID` from each device and create a list with the values. Add this list to the `MachineConfig` object file. The `MachineConfig` Operator generates the `/etc/modprobe.d/vfio.conf` on the nodes with the PCI devices, and binds the PCI devices to the VFIO driver.
+
+.Prerequisites
+* Adding kernel arguments to enable IOMMU for the CPU.
+
+.Procedure
+. Create the `MachineConfig` object file to enable the binding of PCI devices to the VFIO driver.
++
+.Example
+[source,yaml]
+----
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig <1>
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: worker <2>
+  name: 100-worker-vfiopci-configuration <3>
+spec:
+  config:
+    ignition:
+      version: 3.2.0
+    storage:
+      files:
+      - contents:
+          source: data:text/plain;charset=utf-8;base64,b3B0aW9ucyB2ZmlvLXBjaSBpZHM9MTBkZToxZWI4Cg== <4>
+        mode: 420 <5>
+        overwrite: true
+        path: /etc/modprobe.d/vfio.conf <6>
+      - contents:
+          source: data:text/plain;charset=utf-8;base64,dmZpby1wY2kK <7>
+        mode: 420
+        overwrite: true
+        path: /etc/modules-load.d/vfio-pci.conf <8>
+
+----
+<1> The type of object being configured is `MachineConfig`.
+<2> Applies the new kernel argument only to worker nodes.
+<3> Named to identify where this `MachineConfig` object fits among the other `MachineConfig` objects.
+<4> The `echo "options vfio-pci ids=10de:1eb8" | base64` command creates the encoded text. The `vendor-ID` value (10de) and the `device-ID` value (1eb8) applies to a single device that binds to the VFIO driver. You can add a list of multiple devices for `source` with their vendor and device information.
+<5> The values specified for `mode` determine the permissions for the file.
+<6> The `/etc/modprobe.d/vfio.conf` was created by the `echo "options vfio-pci ids=10de:1eb8 > /etc/modprobe.d/vfio/conf"` command.
```

Comment:
I'm not sure why, but the `>` renders weirdly (it looks like a triangle above a `>`) and everything after "echo" is bold in the rendering.